src/Polly.Contrib.DuplicateRequestCollapser.Specs/RequestCollapserTResultSpecs.cs

```diff
@@ -7,11 +8,16 @@ public class RequestCollapserTResultSpecs : RequestCollapserTResultSpecsBase
     {
         public RequestCollapserTResultSpecs(ITestOutputHelper testOutputHelper) : base(testOutputHelper) { }
 
+        private ConcurrentDictionary<(bool, IKeyStrategy, ISyncLockProvider), IsPolicy> PolicyCache = new ConcurrentDictionary<(bool, IKeyStrategy, ISyncLockProvider), IsPolicy>();
+
         protected override IsPolicy GetPolicy(bool useCollapser, IKeyStrategy overrideKeyStrategy = null, ISyncLockProvider lockProvider = null)
         {
-            return useCollapser ?
-                RequestCollapserPolicy<ResultClass>.Create(overrideKeyStrategy ?? RequestCollapserPolicy.DefaultKeyStrategy, lockProvider ?? RequestCollapserPolicy.GetDefaultLockProvider())
-                : (ISyncPolicy<ResultClass>)Policy.NoOp<ResultClass>();
+            return PolicyCache.GetOrAdd((useCollapser, overrideKeyStrategy, lockProvider), _ =>
+            {
```

Comment:
Is this right?  Don't we want to create a new policy for every test?

Reply:
I believe XUnit will run every test in isolation so it will reconstruct this object for each test thereby creating a new policy for each test.

Reply:
Perfect!